src/main/java/com/netchill/api/moviedb/MovieDBApiService.java

```diff
@@ -0,0 +1,84 @@
+package com.netchill.api.moviedb;
+
+import com.netchill.services.configuration.ConfigurationService;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.Getter;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import retrofit2.Call;
+import retrofit2.Retrofit;
+import retrofit2.converter.jackson.JacksonConverterFactory;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import javax.ws.rs.core.HttpHeaders;
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+@Singleton
+public class MovieDBApiService {
```

Comment:
`TmdbApiClient` à la place.

**Les services API:** contiennent les appels API ainsi que la logique liée à l'API.
**Les client API:** fournissent les méthodes permettant de communiquer avec l'API, ils sont utilisés par les services API

Reply:
Ou encore mieux, on pourrait en faire un builder.

1. renommer en `TmdbApiClientBuilder`
2. retirer le @Getter sur `retrofitClient`
3. créer une méthode:
```
build(final Class<T> retrofitClass) {
    return this. retrofitClient.create(retrofitClass)
}
```
4. dans les services api remplacer:
`this.apiClient.getRetrofitClient().create(MovieApiRetrofit.class);`
Par
`this.apiClient.build(MovieApiRetrofit.class);`


Tu en penses quoi ? on peut en discuter si besoin !